### DISCUSSION TITLE
[Feature]: clarification on AlertDialog vs Dialog

### DISCUSSION THREAD

**Discussion by maelp** [unovue/reka-ui#764] [Q&A]
### Describe the feature

I understand that AlertDialog doesn't let the user click outside to close, so it forces the user to choose an option. I'm wondering whether the underlying implementation is so different from a Dialog that it couldn't simply be implemented with eg something like

```
<Dialog mode="alert">
```

to have the same behavior between Dialog and AlertDialog

this might help us develop more features where we want to present content in a modal and let the user choose between a Dialog or an AlertDialog easily, rather than having to do two different implementations in order to do this

### Additional information

- [ ] I intend to submit a PR for this feature.
- [ ] I have already implemented and/or tested this feature.

**Comment by zernonia** [comment]
Thanks for the question @maelp ! A few answers here.

1. We adapt to the component designed in original [Radix UI](https://www.radix-ui.com/primitives/docs/components/alert-dialog)
2. `AlertDialog` is built on top of `Dialog`, with a few core (a11y) behavior difference.
3. `AlertDialog` is built and expose to fit the [w3c's Alert Dialog section](https://www.w3.org/WAI/ARIA/apg/patterns/alertdialog/), as oppose to [Dialog](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/)


Technically you can build your own `AlertDialog` feature based on `Dialog`, but we provide a ready-to-use component here.

**Comment by ianpaschal** [comment]
Good points @zernonia, I think it's still a valid question but maybe @maelp should repost this in the core Radix library. I do agree it would make sense to simply add those extra features (points 2 and 3) to the Dialog with a flag.

**Comment by alecfilios** [comment]
any news on that?
